### DISCUSSION TITLE
What is the difference between nbFramesF and nbSectionsS in Cosserat plugin examples?

### DISCUSSION THREAD

**Discussion by Junang-Wang** [sofa-framework/sofa#4414] [Cosserat] [answered]
Hello,
I am a beginner and I am using the Cosserat plugin in SOFA. I understand that the Cosserat plugin discretizes a continuous rod into several parts. However, I am confused about the two variables nbFramesF and nbSections for discretizing the rod. What do these two variables represent? Why are two variables needed to discretize the rod?

Thank you for your help.

**Comment by adagolodjo** [comment, marked as answer]
Hi @Junang-Wang, 

Welcome to the SOFA community!

Yes, in order to model your beam, you will need to specify these two parameters: nbSections and nbFramesF. The first parameter, nbSections, determines the number of segments of your beam in the local frame. The second parameter, nbFramesF, defines the number of frames of the virtual beam in the global frame. This second parameter has more of a geometric effect, as it helps to smooth the deformation of the beam. I have a tutorial on the "_actuatorBranch_" branch of the repository that you can refer to if you want to understand the principle better (**docs/text/Setting up the Environment.md**). To start, you can set these two parameters to the same value.

Best,
